service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentsController.kt

```diff
@@ -82,19 +97,23 @@ class AttachmentsController(
         return ResponseEntity.ok(id)
     }
 
-    @PostMapping("/citizen", consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
+    @PostMapping(
+        value = ["/citizen/income-statements/{incomeStatementId}", "/citizen/income-statements"],
+        consumes = [MediaType.MULTIPART_FORM_DATA_VALUE]
+    )
     fun uploadAttachmentCitizen(
         db: Database,
         user: AuthenticatedUser,
+        @PathVariable(required = false) incomeStatementId: IncomeStatementId?,
         @RequestPart("file") file: MultipartFile
-    ): ResponseEntity<AttachmentId> {
+    ): AttachmentId {
         Audit.AttachmentsUpload.log(targetId = "nothing")
         user.requireOneOfRoles(UserRole.END_USER)
 
         checkAttachmentCount(db, user)
```

Comment:
We should probably not share the application attachment count limit here.

Reply:
However, we should check access if `incomeStatementId` is provided, right?

Reply:
Access check added. Changed count limiting to allow `maxAttachments` per "parent entity" (application, income statement, no parent).